src/ShaderGen.Tests/AutoGenerated/BuiltinsTests.cs

```diff
@@ -0,0 +1,359 @@
+        /// Initializes a new instance of the <see cref="BuiltinsTests"/> class.
+        /// </summary>
+        /// <param name="output">The output.</param>
+        public BuiltinsTests(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [SkippableFact(typeof(RequiredToolFeatureMissingException), Skip = SkipReason)]
+        private void TestBuiltins()
+        {
+            // Find all backends that can create a headless graphics device on this system.
+            IReadOnlyList<ToolChain> toolChains = ToolChain.Requires(ToolFeatures.HeadlessGraphicsDevice, false);
+            if (toolChains.Count < 1)
+            {
+                throw new RequiredToolFeatureMissingException(
+                    $"At least one tool chain capable of creating headless graphics devices is required for this test!");
+            }
+
+            string csFunctionName = "ComputeShader.CS";
+
+            /*
+             * Auto-generate C# code for testing methods.
+             */
+            IReadOnlyList<MethodInfo> methods = null;
+            Mappings mappings;
+            Compilation compilation;
+            using (new TestTimer(_output, () => $"Generating C# shader code to test {methods.Count} methods"))
+            {
+                // Get all the methods we wish to test
+                methods = MethodsToTest.ToArray();
+                mappings = CreateMethodTestCompilation(methods, out compilation);
+
+                // Note, you could use compilation.Emit(...) at this point to compile the auto-generated code!
+                // however, for now we'll invoke methods directly rather than executing the C# code that has been
+                // generated, as loading emitted code into a test is currently much more difficult.
+            }
+
+            // Allocate enough space to store the result sets for each backend!
+            TestSets testSets = null;
+            using (new TestTimer(
+                _output,
+                () => $"Generating random test data ({(mappings.BufferSize * testSets.TestLoops).ToMemorySize()}) for {testSets.TestLoops} iterations of {mappings.Methods} methods")
+            )
+            {
+                testSets = new TestSets(toolChains, compilation, mappings, TestLoops, MinMantissa, MaxMantissa);
+            }
+
+            /*
+             * Transpile shaders
+             */
+
+            ShaderGenerationResult generationResult;
+
+            using (new TestTimer(
+                _output,
+                t =>
+                    $"Generated shader sets for {string.Join(", ", toolChains.Select(tc => tc.Name))} backends in {t * 1000:#.##}ms.")
+            )
+            {
+                ShaderGenerator sg = new ShaderGenerator(
+                    compilation,
+                    testSets.Select(t => t.Backend).Where(b => b != null).ToArray(),
+                    null,
+                    null,
+                    csFunctionName);
```

Comment:
The "function name" used here should actually just be the function name alone (not including type), because this is passed through to Veldrid. So it should just be "CS". This causes the Metal tests to fail.

Reply:
Changing to just "CS" threw:

> ShaderGen.ShaderGenerationException : The name passed to computeFunctionName must be a fully-qualified type and method.
   at ShaderGen.ShaderGenerator..ctor(Compilation compilation, LanguageBackend[] languages, String vertexFunctionName, String fragmentFunctionName, String computeFunctionName, IShaderSetProcessor[] processors) in D:\Source Control\ShaderGen\src\ShaderGen\ShaderGenerator.cs:line 126
   at ShaderGen.Tests.AutoGenerated.BuiltinsTests.TestBuiltins() in D:\Source Control\ShaderGen\src\ShaderGen.Tests\AutoGenerated\BuiltinsTests.cs:line 149

So this may be a problem.

Reply:
Sorry, maybe I commented on the wrong place. ShaderGen indeed wants the fully-qualified name, because it needs all that info to locate the function. When you pass the actual shader code to Veldrid, on the other hand, you should just pass the actual function name ("CS"). I thought the line I commented on was the variable that ultimately goes into Veldrid.

Reply:
Got you, I just spotted that and have fixed in latest push.